### PR TITLE
Add `dryRun` option

### PR DESCRIPTION
**Pull request by DiamondeX**
Added test option just to see files and folders that could be deleted without actual deletion

### CODE REVIEW

**Comment by SamVerschueren** [comment]
Not sure about this, not really a feature that belongs here. If you want this, maybe create a new repository, something like `del-test`.

**Comment by DiamondeX** [comment]
Well, let owner say he's word.
Meanwhile, could you or someone else help me understand reason the build failed.

**Comment by schnittstabil** [comment]
I can well imagine the benefit of a `-n, --dry-run` CLI option – but I'm not sure if there is an use case for an API option.
@DiamondeX Do you have a concrete use case?

**Comment by schnittstabil** [comment]
@DiamondeX Just run `npm test`:
```
/home/travis/build/sindresorhus/del/index.js
  39:6  error  Keyword "if" must be followed by whitespace  space-after-keywords
  65:5  error  Keyword "if" must be followed by whitespace  space-after-keywords
```

**Comment by DiamondeX** [comment]
thank you, @schnittstabil now I see myself, it's because of lack of experience with travis.
I decide to add this option for complicated cases, with negation pattern involved. To be sure the tool is doing what you realy want.
@schnittstabil you said about CLI option - do you mean that `del` could be used with CLI?

**Comment by schnittstabil** [comment]
>  do you mean that del could be used with CLI?

I'm not aware of any CLI project just wrapping `del` – the readme suggest [trash-cli](https://github.com/sindresorhus/trash-cli) instead.

File: index.js [hunk 1]
```diff
@@ -46,13 +53,18 @@ module.exports.sync = function (patterns, opts) {
...
+	delete opts.test;
+
 	return globby.sync(patterns, opts).map(function (file) {
 		if (!force) {
 			safeCheck(file);
 		}
 
 		file = path.resolve(opts.cwd || '', file);
-		rimraf.sync(file);
+		if(!test) {
```
Comment (schnittstabil) [inline index.js]: Please insert a line break before the `if (!test) {`, to be consistent with the `if (test) {` at line 39.

**Comment by sindresorhus** [comment]
@schnittstabil I have a CLI for it locally. Just haven't had the time to polish it and publish. Planning to do it soon. This could indeed be useful there.

**Comment by DiamondeX** [comment]
Yet another reason: sometimes dependencies may change their behavior slightly, but with large consequences in particular cases for use cases of dependent projects. And one project could lose sync with other. The `test` option ensure you will get what you expect.

File: index.js (hunk 1 above)
Comment (DiamondeX) [inline index.js]: @schnittstabil, I will need to create new pull request or I can do this within this one?

Comment (sindresorhus) [inline index.js]: @DiamondeX You do it within this one. Just push new commits to this pull request.

**Comment by sindresorhus** [comment]
I think `dryRun` would be a better option name. @SamVerschueren @schnittstabil Thoughts?

**Comment by DiamondeX** [comment]
@sindresorhus No problem, I will fix option name, if it's required

**Comment by schnittstabil** [comment]
`dryRun` would be nice, especially because [meow](https://github.com/sindresorhus/meow) and similar tools transform `--dry-run` to `dryRun`:
```js
> meow({argv: ['--dry-run']}).flags
{ dryRun: true }
```

**Comment by DiamondeX** [comment]
ok, i'm fixing now...

**Comment by DiamondeX** [comment]
I'm changing files one by one right from browser and it's required separate commit for each file. Is there a way to change more than one file in one commit from browser?

**Comment by schnittstabil** [comment]
> Is there a way to change more than one file in one commit from browser?

I don't think so.

@DiamondeX Only tests for your new option are missing. Please take a look at `test.js`.

**Comment by DiamondeX** [comment]
@schnittstabil, ok, just a minute

**Comment by schnittstabil** [comment]
> @schnittstabil, ok, just a minute

There is no need to rush.

**Comment by schnittstabil** [comment]
@sindresorhus lgtm

**Comment by sindresorhus** [comment]
Thank you @DiamondeX :)

**Comment by sindresorhus** [comment]
CLI done → https://github.com/sindresorhus/del-cli :)

**Pull request closed without merging**